core/modules/replica/IngestFileSvc.cc

```diff
@@ -95,11 +96,11 @@ void IngestFileSvc::openFile(TransactionId transactionId,
     try {
         auto transactionInfo = _serviceProvider->databaseServices()->transaction(_transactionId);
         if (transactionInfo.state != TransactionInfo::STARTED) {
-            throw logic_error(context_ + "transaction " + to_string(_transactionId) + " is not active");
+            retryAllowedError(context_, "transaction " + to_string(_transactionId) + " is not active");
```

Comment:
I'm not sure I understand this. If client retries this again then result will be the same error unless transaction is started somehow. It feels like you are probably abusing "retry" to mean "do not kill this transaction but correct some other condition first" which is different from "retry immediately". Maybe you need separate error to indicate that transaction is safe to continue but error is an error? This applies to other cases in this file.

Reply:
In theory you're right. There is no way for an ingest workflow to determine if the transaction needs to be started ... unless the client will parse the error message. And the later is very impractical. Introducing another error code to elaborate the problem may not help ether since most ingest workflows may do transaction management in a separate process from the one that's initiated the actual ingest. So, perhaps, the best course of action here would be to roll back the change to the previous code.